pkg/registry/pod/strategy.go

```diff
@@ -65,6 +65,21 @@ func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newPod := obj.(*api.Pod)
 	oldPod := old.(*api.Pod)
 	newPod.Status = oldPod.Status
+	// TODO: VolumeMounts of init containers currently aren't merged properly
+	// by the PATCH handler because they are stored as a string under
+	// PodInitContainersAnnotationKey (pod.alpha.kubernetes.io/init-containers)
+	// in annotations of v1.PodSpec.
+	// In the existing Pod, VolumeMounts of init containers may be altered by
+	// secrets / config maps mechanisms causing ValidatePodUpdate to fail.
+	// As a workaround, changes in VolumeMounts of init containers are
+	// currently ignored and don't cause validation errors.
+	// Remove this workaround when InitContainers go out of alpha and
+	// PodInitContainersAnnotationKey is removed.
+	if len(oldPod.Spec.InitContainers) == len(newPod.Spec.InitContainers) {
```

Comment:
Other parts of this PR address problems that are **not** limited to alpha and will **not** resolve by themselves once init containers leave alpha state. They will be needed in any case, otherwise apply will not work on init containers. But without the workaround there will be no immediate effect of the fix with apply on pods w/init containers remaining broken for now and the new e2e test failing.